### PR TITLE
[Filebeat] AWS S3 input split JSON arrays into multiple events

### PR DESCRIPTION
**Pull request by legoguy1000**
## What does this PR do?

Mirror the feature of the Azure Event Hub input that if the message is a JSON array of objects, split it into multiple events

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have made corresponding change to the default configuration files
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

- Relates #29951

### CODE REVIEW

**Comment by aspacca** [comment]
@legoguy1000 , this would be a breaking change as it is :)

my personal opinion is that it should be a feature behind a config flag in order to enable it

@kaiyan-sheng  , @andrewkroh what do you think?

**Comment by legoguy1000** [comment]
As I was adding the change log I had an inclination that it could be a breaking change.

**Comment by andrewkroh** [comment]
I do think an explicit configuration is needed to avoid a breaking change. How about requiring this to split an array of objects (inspired by `jq`): 

`expand_event_list_from_field: .[]`

**Comment by aspacca** [comment]
> I do think an explicit configuration is needed to avoid a breaking change

yes, as long as the behaviour does not apply automatically, it's not needed.
your proposal with `expand_event_list_from_field: .[]` is fine for me

File: x-pack/filebeat/input/awss3/s3_objects.go [hunk 1]
```diff
@@ -208,26 +209,65 @@ func (p *s3ObjectProcessor) readJSON(r io.Reader) error {
...
+	if err != nil {
+		// return entire message
+		p.log.Debugf("deserializing multiple messages to an array returning error: %s", err)
+		return append(messages, string(bMessage))
+	}
+	p.log.Debugf("deserializing multiple messages to an array")
+	for _, ms := range arrayObject {
+		js, err := json.Marshal(ms)
+		if err != nil {
+			p.log.Errorw(fmt.Sprintf("serializing message %s", ms), "error", err)
```
Comment (rdner) [inline x-pack/filebeat/input/awss3/s3_objects.go:264]: Same here, we should not expose customer's raw data in logs.

**Review by rdner — CHANGES_REQUESTED** [review]

**Comment by andrewkroh** [comment]
Closing because https://github.com/elastic/beats/pull/35475 implemented this.

**Comment by jlind23** [comment]
Closing according to @andrewkroh's comment.

**Pull request closed without merging**